pkgs/top-level/all-packages.nix

```
@@ -7063,6 +7063,8 @@ with pkgs;
 
   hockeypuck-web = callPackage ../servers/hockeypuck/web.nix { };
 
+  holochain-launcher-bin = callPackage ../applications/misc/holochain-launcher-bin { };
```

```suggestion
  holochain-launcher = callPackage ../applications/misc/holochain-launcher { };
```

Please rename the directory to `holochain-launcher`

---

i do plan on packaging the holochain-launcher as a source build in the future. i would prefer if the source build package is named holochain-launcher.

---

also, my intention is to have both packages in the tree simultaneously.

---

@WolfangAukang curious for your feedback on the above

---

My question would be what is the difference between the source and the binary version? This is to understand why having both versions on Nixpkgs. I know this is common practice on AUR, but I don't know any similar case here.

---

an example that i'm aware of is firefox vs firefox-bin.

the release binaries of the holochain-launcher currently ship with multiple vendored binaries of holochain and also a lair-keystore binary. building this default bundle from source has significant resource implications. OTOH, the source build can be configurable and users could choose to opt-out of (some of the) default holochain binaries.

i also think that the source build will be more prone to breaking and thus it can be beneficial to have the binary package as well in cases someone favors most recent versions over the advantages of the source build.

---

@SuperSandro2000 please read the conversation here some of your suggestions depend on the outcome of it.

---

> building this default bundle from source has significant resource implications.

but we would build them in separate packages? Then they are cached by hydra and for an end user it makes no difference. 

> i also think that the source build will be more prone to breaking and thus it can be beneficial to have the binary package as well in cases someone favors most recent versions over the advantages of the source build.

The launcher at least is build on rust and javascript which is unlikely to break. Also nixpkgs already has a very fast speed for updates so I don't think that update speed is a real world concern.

The main problems I see:
- end users do not know which package to choose
- more maintenance
- having both packages uses more resources than only having one.

---

> The launcher at least is build on rust and javascript which is unlikely to break.

it's built on tauri, which does rely on rust and javascript but has its own build process. my initial attempt of packaging it revealed unsolvable issues in tauri's toolchain which i did report upstream and are probably solved by now (haven't tested it yet).

> Also nixpkgs already has a very fast speed for updates so I don't think that update speed is a real world concern.

just to clarify my concern is not about the auto-update involving the bot, but rather repairing breaking changes in upstream packaging.

however, your final points resonate with me enough that i'm willing to consider the binary package as a stop-gap:


> The main problems I see:
> 
>     * end users do not know which package to choose
>     * more maintenance
>     * having both packages uses more resources than only having one.